--- Trilinos-trilinos-release-12-14-1/packages/PyTrilinos/src/Tpetra.i	2019-02-27 05:42:22.000000000 -0800
+++ patch-files/Tpetra.i	2019-03-11 17:04:22.421275025 -0700
@@ -81,6 +81,9 @@
 
 // Tpetra include files
 #include "PyTrilinos_Tpetra_Headers.hpp"
+using Tpetra::Map;
+using Tpetra::Export;
+using Tpetra::Import;
 
 #ifdef HAVE_DOMI
 // Domi include files
@@ -788,11 +791,11 @@
   }
 }
 
-%ignore Tpetra::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::Map;
-%ignore Tpetra::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getLocalElement;
-%ignore Tpetra::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getGlobalElement;
-%ignore Tpetra::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getRemoteIndexList;
-%ignore Tpetra::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getMyGlobalIndices;
+%ignore Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::Map;
+%ignore Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getLocalElement;
+%ignore Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getGlobalElement;
+%ignore Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getRemoteIndexList;
+%ignore Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD, PYTRILINOS_GLOBAL_ORD, DefaultNodeType >::getMyGlobalIndices;
 
 %include "Tpetra_Map_decl.hpp"
 
@@ -800,21 +803,23 @@
 // directives below are redundant, because it is the same as the
 // default template argument.  But SWIG is much more acurate when
 // comparing types when all template arguments are specified.
-%teuchos_rcp(Tpetra::Map< PYTRILINOS_LOCAL_ORD,
-                          PYTRILINOS_GLOBAL_ORD,
-                          DefaultNodeType >)
-%template(Map_default) Tpetra::Map< PYTRILINOS_LOCAL_ORD,
-                                    PYTRILINOS_GLOBAL_ORD,
-                                    DefaultNodeType >;
+
+%teuchos_rcp(Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD,
+                                   PYTRILINOS_GLOBAL_ORD,
+                                   DefaultNodeType >)
+%template(Map_default) Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD,
+                                             PYTRILINOS_GLOBAL_ORD,
+                                             DefaultNodeType >;
+
 %pythoncode
 {
 Map = Map_default
 }
 %inline
 %{
-  typedef Tpetra::Map< PYTRILINOS_LOCAL_ORD,
-                       PYTRILINOS_GLOBAL_ORD,
-                       DefaultNodeType > DefaultMapType;
+ typedef Tpetra::Classes::Map< PYTRILINOS_LOCAL_ORD,
+                                PYTRILINOS_GLOBAL_ORD,
+                                DefaultNodeType > DefaultMapType;
 %}
 
 /////////////////////////////
@@ -866,12 +871,12 @@
 // Tpetra Export support //
 ///////////////////////////
 %include "Tpetra_Export_decl.hpp"
-%teuchos_rcp(Tpetra::Export< PYTRILINOS_LOCAL_ORD,
-                             PYTRILINOS_GLOBAL_ORD,
-                             DefaultNodeType >)
-%template(Export_default) Tpetra::Export< PYTRILINOS_LOCAL_ORD,
-                                          PYTRILINOS_GLOBAL_ORD,
-                                          DefaultNodeType >;
+%teuchos_rcp(Tpetra::Classes::Export< PYTRILINOS_LOCAL_ORD,
+                                      PYTRILINOS_GLOBAL_ORD,
+                                      DefaultNodeType >)
+%template(Export_default) Tpetra::Classes::Export< PYTRILINOS_LOCAL_ORD,
+                                                   PYTRILINOS_GLOBAL_ORD,
+                                                   DefaultNodeType >;
 %pythoncode
 {
 Export = Export_default
@@ -881,12 +886,12 @@
 // Tpetra Import support //
 ///////////////////////////
 %include "Tpetra_Import_decl.hpp"
-%teuchos_rcp(Tpetra::Import< PYTRILINOS_LOCAL_ORD,
-                             PYTRILINOS_GLOBAL_ORD,
-                             DefaultNodeType >)
-%template(Import_default) Tpetra::Import< PYTRILINOS_LOCAL_ORD,
-                                          PYTRILINOS_GLOBAL_ORD,
-                                          DefaultNodeType >;
+%teuchos_rcp(Tpetra::Classes::Import< PYTRILINOS_LOCAL_ORD,
+                                      PYTRILINOS_GLOBAL_ORD,
+                                      DefaultNodeType >)
+%template(Import_default) Tpetra::Classes::Import< PYTRILINOS_LOCAL_ORD,
+                                                   PYTRILINOS_GLOBAL_ORD,
+                                                   DefaultNodeType >;
 %pythoncode
 {
 Import = Import_default
